test/test_packages.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+"""
+test_packages
+~~~~~~~~~~~~~~~
+This test module tests if R and Python packages installed can be imported.
+It's a basic test aiming to prove that the package is working properly.
+
+The goal is to detect import errors that can be caused by incompatibilities between packages for example:
+
+- #1012: issue importing `sympy`
+- #966: isssue importing `pyarrow`
+
+This module checks dynmamically, through the `CondaPackageHelper`, only the specified packages i.e. packages requested by `conda install` in the `Dockerfiles`.
+This means that it does not check dependencies. This choice is a tradeoff to cover the main requirements while achieving reasonabe test duration.
+However it could be easily changed (or completed) to cover also dependencies `package_helper.installed_packages()` instead of `package_helper.specified_packages()`.
+
+Example:
+
+    $ make test/datascience-notebook
+
+    # [...]
+    # test/test_packages.py::test_python_packages 
+    # --------------------------------------------------------------------------------------------- live log setup ----------------------------------------------------------------------------------------------
+    # 2020-03-08 09:56:04 [    INFO] Starting container jupyter/datascience-notebook ... (helpers.py:51)
+    # 2020-03-08 09:56:04 [    INFO] Running jupyter/datascience-notebook with args {'detach': True, 'ports': {'8888/tcp': 8888}, 'tty': True, 'command': ['start.sh', 'bash', '-c', 'sleep infinity']} ... (conftest.py:78)
+    # 2020-03-08 09:56:04 [    INFO] Grabing the list of specifications ... (helpers.py:76)
+    # ---------------------------------------------------------------------------------------------- live log call ----------------------------------------------------------------------------------------------
+    # 2020-03-08 09:56:07 [    INFO] Testing the import of packages ... (test_packages.py:125)
+    # 2020-03-08 09:56:07 [    INFO] Trying to import conda (test_packages.py:127)
+    # 2020-03-08 09:56:07 [    INFO] Trying to import notebook (test_packages.py:127)
+    # 2020-03-08 09:56:08 [    INFO] Trying to import jupyterhub (test_packages.py:127)
+    # [...]
+
+"""
+
+import logging
+
+import pytest
+
+from helpers import CondaPackageHelper
+
+LOGGER = logging.getLogger(__name__)
+
+# Mapping between package and module name
+PACKAGE_MAPPING = {
+    # Python
+    "matplotlib-base": "matplotlib",
+    "beautifulsoup4": "bs4",
+    "scikit-learn": "sklearn",
+    "scikit-image": "skimage",
+    # R
+    "randomforest": "randomForest",
+    "rsqlite": "DBI",
+    "rcurl": "RCurl",
+    "rodbc": "RODBC",
+}
+
+# List of packages that cannot be tested in a standard way
+EXCLUDED_PACKAGES = [
+    "tini",
+    "python",
+    "hdf5",
+    "conda-forge::blas[build",
+    "protobuf",
+    "r-irkernel",
+    "unixodbc",
+    "spylon-kernel",
```

Comment:
You should be able to `import spylon_kernel` to test this one. https://github.com/Valassis-Digital-Media/spylon-kernel#using-it-as-an-ipython-magic

Reply:
You're right just need a specific mapping.

```python
# Mapping between package and module name
PACKAGE_MAPPING = {
    # Python
    "spylon-kernel": "spylon_kernel",
```